Source/OxyPlot.SkiaSharp/SkiaRenderContext.cs

```diff
@@ -0,0 +1,762 @@
+                    return this.Convert(rect);
+                default:
+                    var actualThickness = this.GetActualThickness(strokeThickness, edgeRenderingMode);
+                    var snapOffset = GetSnapOffset(actualThickness);
+                    return this.ConvertSnap(rect, snapOffset);
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="SKRect"/>s that should actually be rendered from the list of <see cref="OxyRect"/>s, taking into account DPI scaling and snapping if necessary.
+        /// </summary>
+        /// <param name="rects">The rectangles.</param>
+        /// <param name="strokeThickness">The stroke thickness.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The actual rectangles.</returns>
+        private IEnumerable<SKRect> GetActualRects(IEnumerable<OxyRect> rects, double strokeThickness, EdgeRenderingMode edgeRenderingMode)
+        {
+            switch (edgeRenderingMode)
+            {
+                case EdgeRenderingMode.PreferGeometricAccuracy:
+                case EdgeRenderingMode.PreferSpeed:
+                    return rects.Select(this.Convert);
+                default:
+                    var actualThickness = this.GetActualThickness(strokeThickness, edgeRenderingMode);
+                    var snapOffset = GetSnapOffset(actualThickness);
+                    return rects.Select(rect => this.ConvertSnap(rect, snapOffset));
+            }
+        }
+
+        /// <summary>
+        /// Gets the stroke thickness that should actually be used for rendering, taking into account DPI scaling and snapping if necessary.
+        /// </summary>
+        /// <param name="strokeThickness">The stroke thickness.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The actual stroke thickness.</returns>
+        private float GetActualThickness(double strokeThickness, EdgeRenderingMode edgeRenderingMode)
+        {
+            var scaledThickness = this.Convert(strokeThickness);
+            if (edgeRenderingMode == EdgeRenderingMode.PreferSharpness)
+            {
+                scaledThickness = Snap(scaledThickness, 0);
+            }
+
+            return scaledThickness;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="SKPaint"/> containing information needed to render the fill of a shape.
+        /// </summary>
+        /// <param name="fillColor">The fill color.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The paint.</returns>
+        private SKPaint GetFillPaint(OxyColor fillColor, EdgeRenderingMode edgeRenderingMode)
+        {
+            this.paint.Color = fillColor.ToSKColor();
+            this.paint.Style = SKPaintStyle.Fill;
+            this.paint.IsAntialias = this.ShouldUseAntiAliasing(edgeRenderingMode);
+            return this.paint;
```

Comment:
It doesn't appear to make any difference in testing, and I can't work out from the documentation if `PathEffect` ever applies to Fills, but should the `PathEffect` be cleared here?